cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -380,6 +450,150 @@ writeLedgerState mOutFile qState@(SerialisedLedgerState serLedgerState) =
      first (const ls) (decodeFull ls)
 
 
+-- | This local data structure is introduced purely to allow nicely formatted output and has no other purpose.
+-- mark, set, go are the three ledger state stake snapshots (most recent to least recent)
+-- go is the snapshot that is used for the current epoch, set will be used in the next epoch,
+-- mark for the epoch after that.  marktot etc record the total active stake for each snapshot
+-- This information can be used by community tools to calculate upcoming leader schedules
+data Stakes =  Stakes {
+      mark, set, go :: Integer,
+      marktot, settot, gotot :: Integer
+    } deriving Show
+
+-- | Pretty printing for stake information
+instance ToJSON Stakes where
+    toJSON (Stakes m s g mt st gt) =
+        object ["poolStakeMark" .= m, "poolStakeSet" .= s, "poolStakeGo" .= g,
+                "activeStakeMark" .= mt, "activeStakeSet" .= st, "activeStakeGo" .= gt]
+
+    toEncoding  (Stakes m s g mt st gt) =
+        pairs ("poolStakeMark" .= m <> "poolStakeSet" .= s <> "poolStakeGo" .= g <>
+               "activeStakeMark" .= mt <> "activeStakeSet" .= st <> "activeStakeGo" .= gt)
+
+
+writeStakeSnapshot :: forall era ledgerera.
+                      ShelleyLedgerEra era ~ ledgerera
+                   => Era.Era ledgerera
+                   => FromCBOR (LedgerState era)
+                   => Hash StakePoolKey
+                   -> SerialisedLedgerState era
+                   -> ExceptT ShelleyQueryCmdError IO ()
+
+writeStakeSnapshot poolId qState =
+  case decodeLedgerState qState of
+    Left bs -> firstExceptT ShelleyQueryCmdHelpersError $ pPrintCBOR bs
+    Right ledgerState ->
+      if maybehk == Nothing then
+        left ShelleyQueryCmdPoolIdError
+      else
+        liftIO . LBS.putStrLn $  encodePretty $ Stakes {mark = markStake,set = setStake,go = goStake,marktot = markTotal,settot = setTotal,gotot = goTotal}
+      where
+        -- Ledger State
+        (LedgerState snapshot)  = ledgerState
+
+        -- The three stake snapshots, obtained from the ledger state
+        (SnapShots markS setS goS _) = esSnapshots $ nesEs snapshot
+
+        -- Calculate the three pool and active stake values for the given pool
+        markStake =   getPoolStake hk markS
```

Comment:
There is code alignment and some irregular spacing in this where clause. We don't align code in the cli (changes whilst keeping the alignment results in semantically null changes). Let's remove the alignment and fix the spacing before and after some of the `=`s in this clause.